server/connection_manager.py

```diff
@@ -100,7 +100,7 @@ def update_code_cache(self, room_code: str, replace_data: ReplaceData) -> None:
                 self._rooms[room_code]["code"] = updated_code
 
     async def broadcast(
-        self, data: SyncData | ReplaceData, room_code: str, sender: Client | None = None, buggy: bool = False
+        self, data: type[EventData], room_code: str, sender: Client | None = None, buggy: bool = False
```

Comment:
`broadcast` still accepts a lot of parameters. As I said in a previous comment, I probably wouldn't leave all this work to `broadcast`. shouldn't this happen in the event handler?

Reply:
Can you elaborate? What shouldn't be in the broadcast?

Reply:
> Can you elaborate? What shouldn't be in the broadcast?

In my opinion `broadcast` should only send the message to every client in a room, in this case, it is also modifying the code and updating the code cache. I would leave that to the event handler.

Reply:
Right, the code would still be moved out of `broadcast` though, but I think it's fine if we leave that out for another PR.